lib/grammers-client/src/types/chat/user.rs

```diff
@@ -8,6 +8,44 @@
 use grammers_tl_types as tl;
 use std::fmt;
 
+/// Platform Identifier.
+#[non_exhaustive]
+pub enum Platform {
+    All,
+    Android,
+    IOS,
+    WindowsPhone,
+    Other(String),
+}
+
+/// Contains the reason why a certain user is restricted.
+pub struct RestrictionReason {
+    pub platforms: Vec<Platform>,
+    pub reason: String,
+    pub text: String,
+}
+
+impl RestrictionReason {
+    pub(crate) fn from_raw(reason: &tl::enums::RestrictionReason) -> Self {
+        let tl::enums::RestrictionReason::Reason(reason) = reason;
+        Self {
+            platforms: reason
+                .platform
+                .split("-")
+                .map(|p| match p {
+                    "all" => Platform::All,
+                    "android" => Platform::Android,
+                    "ios" => Platform::IOS,
+                    "WindowsPhone" => Platform::WindowsPhone,
```

Comment:
Where did you get this list of strings from? It would be nice to link it with a comment if possible.